joomla/joomla-cms
# Localise the format for the calendar formfield

With the current calendar we can specify the date format in the XML file. Eg "%Y-%m-%d %H:%M:%S". However that is hardcoded and the same for every language.
This PR proposes that we can set the calendar so it adjusts the format to the active language.
### Summary of Changes

Introduces two new attributes `showtime` and `translateformat` for the JFormFieldCalendar:
- `translateformat`: If set to true, the calendar will use a language string to determine the format. The `format` attribute is ignored. If false, the `format` attribute is used (same behaviour as today).
- `showtime`: If set to true and `translateformat` is true, the language key `DATE_FORMAT_CALENDAR_DATETIME` is used, otherwise `DATE_FORMAT_CALENDAR_DATE`.
  Please note that this attribute is also used in the PR #11138 to specify if the calendar should show a timefield or not. Imho it makes sense to use the same attribute here.
### Testing Instructions
1. Apply PR
2. Add the new language strings to your second language you have installed. Eg for de-DE add this to both de-DE.ini files (frontend and backend):

```
DATE_FORMAT_CALENDAR_DATE="%d.%m.%Y"
DATE_FORMAT_CALENDAR_DATETIME="%d.%m.%Y %H:%M:%S"
```
1. Check backend article edits and watch the format changing from YYYY-MM-DD to DD.MM.YYYY if you change the language.
2. Active the user - profile plugin and edit your profil in frontend. Check the birthday field and watch the change in format. Also this field should **not** show the time.
### Current Limitations
- This PR so far only changed the backend article form and the frontend profile form. If accepted it can of course be expanded to all other forms as well (just needs change in the XML).
- I honestly don't know if this will work for all languages and possible date formats. That is something the language pack creators have to test when creating the language strings.
### Documentation Changes Required

New attributes need to be added to https://docs.joomla.org/Calendar_form_field_type
